src/plonk/circuit.rs

```diff
@@ -47,13 +47,13 @@ impl<C: ColumnType> Ord for Column<C> {
             (Any::Advice, Any::Advice)
             | (Any::Instance, Any::Instance)
             | (Any::Fixed, Any::Fixed) => self.index.cmp(&other.index),
-            // Across column types, sort Advice < Instance < Fixed.
-            (Any::Advice, Any::Instance)
+            // Across column types, sort Instance < Advice < Fixed.
```

Comment:
This does not have any practical effect on the V1 layouter, since instance columns never actually occur within regions, but it keeps the inherent ordering in line with the rendered ordering.